File: internal/infrastructure/http/v1/dto/worker_job.go
package dto

import (
	"metapus/internal/core/workerjob"
)

// WorkerJobResponse is the API representation of a single worker job run.
type WorkerJobResponse struct {
	ID             string  `json:"id"`
	JobName        string  `json:"jobName"`
	JobCategory    string  `json:"jobCategory"`
	Status         string  `json:"status"`
	StartedAt      string  `json:"startedAt"`
	FinishedAt     *string `json:"finishedAt"`
	DurationMs     *int    `json:"durationMs"`
	ItemsProcessed *int    `json:"itemsProcessed"`
	ErrorMessage   *string `json:"errorMessage,omitempty"`
}

// WorkerJobStatsResponse is the API representation of KPI stats.
type WorkerJobStatsResponse struct {
	Total       int64 `json:"total"`
	Success     int64 `json:"success"`
	Error       int64 `json:"error"`
	AvgDuration int64 `json:"avgDuration"` // milliseconds
}

// WorkerJobListResponse is the paginated list response.
type WorkerJobListResponse struct {
	Items      []WorkerJobResponse `json:"items"`
	NextCursor string              `json:"nextCursor,omitempty"`
	HasMore    bool                `json:"hasMore"`
	TotalCount int64               `json:"totalCount"`
}

// MapWorkerJob converts a domain Job to its DTO.
func MapWorkerJob(j workerjob.Job) WorkerJobResponse {
	r := WorkerJobResponse{
		ID:             j.ID.String(),
		JobName:        j.JobName,
		JobCategory:    j.JobCategory,
		Status:         string(j.Status),
		StartedAt:      j.StartedAt.UTC().Format("2006-01-02T15:04:05Z"),
		DurationMs:     j.DurationMs,
		ItemsProcessed: j.ItemsProcessed,
		ErrorMessage:   j.ErrorMessage,
	}
	if j.FinishedAt != nil {
		s := j.FinishedAt.UTC().Format("2006-01-02T15:04:05Z")
		r.FinishedAt = &s
	}
	return r
}

// MapWorkerJobs converts a slice of domain Jobs to DTOs.
func MapWorkerJobs(jobs []workerjob.Job) []WorkerJobResponse {
	out := make([]WorkerJobResponse, len(jobs))
	for i, j := range jobs {
		out[i] = MapWorkerJob(j)
	}
	return out
}

File: frontend/types/worker-job.ts
/** Single worker job execution run from API */
export interface WorkerJob {
  id: string;
  jobName: string;
  jobCategory: string;
  status: WorkerJobStatus;
  startedAt: string;
  finishedAt?: string;
  durationMs?: number;
  itemsProcessed?: number;
  errorMessage?: string;
}

export type WorkerJobStatus = "running" | "success" | "error" | "skipped";

/** Aggregated KPI stats for the last 24 hours */
export interface WorkerJobStats {
  total: number;
  success: number;
  error: number;
  avgDuration: number; // milliseconds
}

/** Paginated list response */
export interface WorkerJobListResponse {
  items: WorkerJob[];
  nextCursor?: string;
  hasMore: boolean;
  totalCount: number;
}

/** Filter options for the list UI */
export interface WorkerJobFilter {
  jobName?: string;
  jobCategory?: string;
  status?: string;
  dateFrom?: string;
  dateTo?: string;
  after?: string;
  limit?: number;
}

// ── UI Constants ──────────────────────────────────────────────────────────

export const JOB_STATUSES: { value: WorkerJobStatus; label: string; color: string }[] = [
  { value: "running",  label: "Выполняется", color: "text-blue-600" },
  { value: "success",  label: "Успешно",     color: "text-green-600" },
  { value: "skipped",  label: "Пропущено",   color: "text-muted-foreground" },
  { value: "error",    label: "Ошибка",      color: "text-red-600" },
]

export const JOB_CATEGORIES: { value: string; label: string }[] = [
  { value: "crypto",     label: "Криптовалюта" },
  { value: "outbox",     label: "Outbox" },
  { value: "cleanup",    label: "Очистка" },
  { value: "automation", label: "Автоматизация" },
]

export const KNOWN_JOB_NAMES: { value: string; label: string }[] = [
  { value: "outbox.relay",                label: "Outbox: отправка" },
  { value: "outbox.recover_stuck",        label: "Outbox: восстановление" },
  { value: "cleanup.sessions",            label: "Очистка: сессии" },
  { value: "cleanup.idempotency",         label: "Очистка: идемпотентность" },
  { value: "cleanup.automation_history",  label: "Очистка: история автоматизации" },
  { value: "cleanup.automation_files",    label: "Очистка: файлы автоматизации" },
  { value: "cleanup.notifications",       label: "Очистка: уведомления" },
  { value: "cleanup.worker_jobs",         label: "Очистка: логи задач" },
  { value: "crypto.expiration",           label: "Крипто: истечение инвойсов" },
  { value: "crypto.sweep_eval",           label: "Крипто: оценка свипа" },
]
